host-services/server.go

```diff
@@ -15,18 +15,34 @@ import (
 )
 
 type HostServicesServer struct {
-	log      *slog.Logger
-	nc       *nats.Conn
-	services map[string]HostService
-	tracer   trace.Tracer
+	log        *slog.Logger
+	ncInternal *nats.Conn
+	services   map[string]HostService
+	// Every single workload gets its own private host services connection,
+	// even if it's reusing defaults for config
+	hsClientConnections map[string]*nats.Conn
```

Comment:
My preference is to put the host services connection in the `AgentClient`, alongside the agent's internal NATS connection. Once we get the tests working we can look at how to make that approach place nicely with the host services server and internal nats structs.

Reply:
Conceptually the host services client doesn't belong inside the agent client, it's the connection through which HostServices makes requests of the built-ins.

Let's get the tests working and we can move forward from there.

Reply:
This makes sense- the idea was born from my desire to move the internal per-workload internal NATS connection into the `AgentClient` and cache it there instead of caching the credentials inside of the NATS internal server struct.